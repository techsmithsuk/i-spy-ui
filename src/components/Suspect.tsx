import React, { useState, useEffect } from 'react';
import fetchData from '../helpers/dataFetcher';


interface SuspectInfo {
    name: string;
}

export default function Suspect() {
    
    const [suspectInfo, setSuspectInfo] = useState<SuspectInfo | null>(null);

    const url = "http://localhost:8080/suspect";

    useEffect(
        () => { 
        fetchData(url)
        .then(jsonResponse => setSuspectInfo(jsonResponse))
    }, [url]);

    if (!suspectInfo){
        return <div>Fetching data...</div>
    }
    
    return <div>{suspectInfo.name}</div>
}