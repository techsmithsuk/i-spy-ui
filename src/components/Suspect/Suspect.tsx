import React, { useState, useEffect } from 'react';
import asyncJSONFetch from '../general/helpers/asyncJSONFetcher';


interface SuspectTrialInfo {
    name: string;
}

export default function SuspectTrial() {

    const [suspectInfo, setSuspectInfo] = useState<SuspectTrialInfo | null>(null);
    const [error, setError] = useState(false);

    const url: string = `${process.env.REACT_APP_API_URL}/suspect_test`
    
    useEffect(() => { 
        asyncJSONFetch(url)            
            .then(jsonResponse => setSuspectInfo(jsonResponse))   
            .catch(() => setError(true))          
    }, [url]);    

    if (error) { 
        return <div>Oh No!!! There was an error</div> 
    }

    if (!suspectInfo){
        return (
            <div>Fetching data...</div>
        )
    }
    
    return ( 
        <div>
            {suspectInfo.name}
        </div>
    )
}