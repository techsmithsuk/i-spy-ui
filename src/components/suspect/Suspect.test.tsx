import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

import Suspect from '../suspect/Suspect';

describe("Suspect", () => {
    it("Should render without errors", () => {
        const component = renderer.create(
            <Suspect/>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

test('renders person card', () => {    
    ReactDOM.render(<Suspect />, document.createElement('Suspect'));
  });