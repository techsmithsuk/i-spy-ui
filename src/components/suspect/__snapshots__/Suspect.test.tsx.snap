// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suspect Should render without errors 1`] = `
<div>
  Fetching data...
</div>
`;
